### ISSUE TITLE
Default dropout behavior

### ISSUE THREAD

**Issue by deep-speech** [rwth-i6/returnn#93]
I have a couple of queries regarding the default dropout behavior in returnn.
Does it also apply to recurrent weights? Does it support variational dropout, if yes then how to use it?

**Comment by deep-speech** [comment]
In the TFUtil.dropout implementation, it checks for dropout_noise_shape. I didn't find any reference for usage of this parameter. What is the significance of this parameter?

**Comment by albertz** [comment]
See `_ConcatInputLayer` how it is used.
The `noise_shape` is the same as in `tf.nn.dropout`.
And it's used in the code (both `TFUtil.dropout` or `tf.nn.dropout`). (Use an editor like PyCharm which can highlight where it is used if you don't find it.)
E.g. consider that you have a tensor of shape (batch, time, dim), and noise shape is (1, 1, dim) (must be compatible for broadcasting), then the dropout mask will also be of this noise shape (1, 1, dim), i.e. the same dropout mask is shared across all seqs in the batch and time frames. This is the default. But just look at the code for details.

**Comment by deep-speech** [comment]
Thanks for clarifying.
